lib/chef/resource/aws_iam_instance_profile.rb

```diff
@@ -0,0 +1,33 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+#
+# An AWS IAM instance profile, a container for an IAM role that you can use to
+# pass role information to an EC2 instance when the instance starts..
+#
+# `name` is unique for an AWS account.
+#
+# API documentation for the AWS Ruby SDK for IAM instance profiles (and the object returned from `aws_object`) can be found here:
+#
+# - http://docs.aws.amazon.com/sdkforruby/api/Aws/IAM/InstanceProfile.html
+#
+class Chef::Resource::AwsIamInstanceProfile < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type ::Aws::IAM::InstanceProfile, id: :name
```

Comment:
Looking at http://docs.aws.amazon.com/sdkforruby/api/Aws/IAM/InstanceProfile.html I think this should be `id: :instance_profile_id`